Source/Fuse.Scripting.JavaScript/JavaScript.Model.uno

```diff
@@ -0,0 +1,254 @@
+using Uno;
+using Uno.UX;
+using Uno.Collections;
+using Uno.Compiler;
+using Fuse.Scripting;
+using Uno.Testing;
+using Uno.Threading;
+using Uno.IO;
+
+namespace Fuse.Reactive
+{
+	//the attached properties need to be in a public class.
+	public partial class JavaScript
+	{
+		class ModelData
+		{
+			public IExpression Model;
+			public NameTable NameTable;
+		}
+		
+		//Requires a NameTable that will be set first.
+        [UXAttachedPropertySetter("JavaScript.Model"), UXNameScope, UXAuxNameTable("ModelNameTable")]
+        public static void SetModel(Visual v, IExpression model)
+        {
+			var md = v.Properties.Get( _modelHandle ) as ModelData;
+			if (md == null)
+			{
+				md = new ModelData{ Model = model };
+				v.Properties.Set( _modelHandle, md );
+			}
+			else
+			{
+				md.Model = model;
+			}
+			
+			Complete( md, v );
+		}
+		
+		static void Complete( ModelData md, Visual v )
+		{
+			v.RemoveAllChildren<ModelJavaScript>();
+			
+			//avoid creating without the NameTable as unfortunately UX will set the Model prior to the NameTable
+			if (md.NameTable == null || md.Model == null)
+				return;
+				
+			var parsed = ModelJavaScript.ParseModelExpression(md.Model, md.NameTable);
+			v.Children.Add( new ModelJavaScript(parsed, md.NameTable, null) );
+        }
+
+        static PropertyHandle _modelHandle = Properties.CreateHandle();
+        
+        [UXAttachedPropertySetterAttribute("ModelNameTable")]
+        public static void SetModelNameTable(Visual v, NameTable nt)
+        {
+			var md = v.Properties.Get( _modelHandle ) as ModelData;
+			if (md == null)
+			{
+				md = new ModelData{ NameTable = nt };
+				v.Properties.Set( _modelHandle, md );
+			}
+			else
+			{
+				md.NameTable = nt;
+			}
+			
+			Complete( md, v );
+        }
+
+        //TODO: This should probably be JavaScript.Model, Preview would need to be adjusted as well
```

Comment:
What's up with this TODO? What implications does leaving this as-is have?

Reply:
I wrote it. It means the property should probably be called "JavaScript.Model" for consistency, but I wasn't involved in the naming to know why that wasn't chosen (maybe something to do with preview).

The typical use-case would work either way `<App Model="blah()"/>` but without the name change this wouldn't work `<App JavaScript.model="blah()"/>` though `<Panel ux:Class="myPanel" JavaScript.Model="blah()"/>` does work.

Reply:
I think we should just change it and fix Preview if it doesn't work there.